server/services/prisonerProfileService/index.test.ts

```diff
@@ -118,6 +132,307 @@ describe('PrisonerProfileService', () => {
     })
   })
 
+  describe('hasAdjudications', () => {
+    it('should return whether the user has adjudications', async () => {
+      const mockUserId = '123456'
+      const mockAgencyId = 'ABC'
+      const mockHasAdjudicationsResponse: HasAdjudicationsResponse = {
+        hasAdjudications: true,
+      }
+
+      hmppsAuthClient.getSystemClientToken.mockResolvedValue(mockToken)
+      adjudicationsApiClientFactory.mockReturnValue(adjudicationsApiClient)
+      adjudicationsApiClient.hasAdjudications.mockResolvedValue(mockHasAdjudicationsResponse)
+
+      const result = await prisonerProfileService.hasAdjudications({
+        idToken: { booking: { id: mockUserId }, establishment: { agency_id: mockAgencyId } },
+      })
+
+      expect(result).toEqual(mockHasAdjudicationsResponse)
+      expect(hmppsAuthClient.getSystemClientToken).toHaveBeenCalled()
+      expect(adjudicationsApiClientFactory).toHaveBeenCalledWith(mockToken)
+      expect(adjudicationsApiClient.hasAdjudications).toHaveBeenCalledWith(mockUserId, mockAgencyId)
+    })
+  })
+
+  describe('getReportedAdjudicationsFor', () => {
+    it('should return reported adjudications for the given user', async () => {
+      const mockBookingId = '123456'
+      const mockAgencyId = 'XYZ456'
+      const mockStatusQueryParam =
+        '&status=ACCEPTED&status=REJECTED&status=AWAITING_REVIEW&status=RETURNED&status=UNSCHEDULED&status=SCHEDULED&status=REFER_POLICE&status=REFER_INAD&status=REFER_GOV&status=PROSECUTION&status=DISMISSED&status=NOT_PROCEED&status=ADJOURNED&status=CHARGE_PROVED&status=QUASHED&status=INVALID_OUTCOME&status=INVALID_SUSPENDED&status=INVALID_ADA'
+
+      const mockReportedAdjudicationsData: PageReportedAdjudicationDto = {
+        content: [],
+      }
+
+      hmppsAuthClient.getSystemClientToken.mockResolvedValue(mockToken)
+      adjudicationsApiClientFactory.mockReturnValue(adjudicationsApiClient)
+      adjudicationsApiClient.getReportedAdjudicationsFor.mockResolvedValue(mockReportedAdjudicationsData)
+
+      const result = await prisonerProfileService.getReportedAdjudicationsFor(mockBookingId, mockAgencyId)
+
+      expect(result).toEqual(mockReportedAdjudicationsData)
+      expect(hmppsAuthClient.getSystemClientToken).toHaveBeenCalled()
+      expect(adjudicationsApiClientFactory).toHaveBeenCalledWith(mockToken)
+      expect(adjudicationsApiClient.getReportedAdjudicationsFor).toHaveBeenCalledWith(
+        mockBookingId,
+        mockAgencyId,
+        mockStatusQueryParam,
+      )
+    })
+  })
+
+  describe('getReportedAdjudication', () => {
+    it('should return the reported adjudication for the given charge number and agency ID', async () => {
+      const mockChargeNumber = 'ABC123'
+      const mockAgencyId = 'XYZ456'
+      const mockReportedAdjudication: ReportedAdjudicationApiResponse = {
```

Comment:
Might make tests easier to read if test data was moved externally, especially when the objects are very large like this?

Reply:
Agreed - that's the plan with the cleanup PR, as I've done exactly that with the transactions and visits code. This one got quite messy and was already quite a big PR when I picked it up from Barry so it definitely needs re-aligning with the other code I've written, but didn't want to make this one any bigger!